compiler/rustc_span/src/lib.rs

```diff
@@ -460,21 +460,21 @@ impl Ord for SpanData {
         let SpanData {
             lo: s_lo,
             hi: s_hi,
-            ctxt: s_ctxt,
+            ctxt: _,
```

Comment:
This needs a comment for macro expansions.
Side note: this `Ord` impl is not consistent with the derived impl for `PartialEq`. Should we fix the latter?

Reply:
What do you think about ensuring that for equal (including the sort-ignored fields) `SpanData`s we just return `Ordering::Equal`, and only sort the others?

Reply:
That's not consistent either.
The `a == b` implies `partial_cmp(a, b) == Some(Equal)` direction is easy. We have it without adding a special case.
The converse is harder: if `s_lo == o_lo && s_hi == o_hi` we get `Some(Equal)`, but `s_ctxt` may not be `o_ctxt`.

Reply:
Maybe `Span`s should just be `PartialOrd`, but not `Ord`. I'll try if we can get away with just removing `Ord`